src/GraphQL.Tests/Types/ObjectGraphTypeTests.cs

```diff
@@ -51,10 +52,21 @@ public void should_throw_on_invalid_graphtype_name()
         }
     }
 
-    [Fact]
-    public void should_ignore_graphqlmetadata_attribute()
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public void should_ignore_graphqlmetadata_attribute(bool useLegacyTypeNaming)
```

Comment:
This test specifically ensured that the GraphQLMetadata attribute was ignored when naming a class.  This is correct; the metadata attribute is not designed for use by graph classes, but only for model classes.  So I've changed the GraphQLName logic to ignore checking for the attribute for graph classes.